.dotnet/src/Custom/Chat/ChatClient.cs

```diff
@@ -200,19 +200,27 @@ public virtual Task<ClientResult<ChatCompletion>> CompleteChatAsync(string messa
         PipelineMessage requestMessage = CreateCustomRequestMessage(messages, choiceCount, options);
         requestMessage.BufferResponse = false;
         Shim.Pipeline.Send(requestMessage);
-        PipelineResponse response = requestMessage.ExtractResponse();
-
-        if (response.IsError)
+        if (requestMessage.Response.IsError)
         {
-            throw new ClientResultException(response);
+            throw new ClientResultException(requestMessage.Response);
         }
 
-        ClientResult genericResult = ClientResult.FromResponse(response);
-        return StreamingClientResult<StreamingChatUpdate>.CreateFromResponse(
-            genericResult,
-            (responseForEnumeration) => SseAsyncEnumerator<StreamingChatUpdate>.EnumerateFromSseStream(
-                responseForEnumeration.GetRawResponse().ContentStream,
-                e => StreamingChatUpdate.DeserializeStreamingChatUpdates(e)));
+        PipelineResponse response = null;
+        try
+        {
+            response = requestMessage.ExtractResponse();
+            ClientResult genericResult = ClientResult.FromResponse(response);
```

Comment:
Do you need this genericResult? It does not seem to be used anywhere.

Reply:
but you are not returning this genericResult to the user, and so they cannot read the headers anyway.